gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -507,7 +535,7 @@ MM_ParallelGlobalGC::mainThreadGarbageCollect(MM_EnvironmentBase *env, MM_Alloca
 #endif /* OMR_GC_MODRON_COMPACTION */
 
 	/* If the delegate has isAllowUserHeapWalk set, fix the heap so that it can be walked */
-	if (_delegate.isAllowUserHeapWalk() || env->_cycleState->_gcCode.isRASDumpGC()) {
+	if (_delegate.isAllowUserHeapWalk() || env->_cycleState->_gcCode.isRASDumpGC() || env->_cycleState->_gcCode.shouldClearHeap()) {
```

Comment:
you can add a comment that clearing heap is another pass that clears all holes and  requires first for dark matter to be converted into holes